Lib/test/test_annotationlib.py

```diff
@@ -1617,6 +1617,32 @@ def test_evaluate_with_type_params_and_scope_conflict(self):
                     TypeParamsSample.TypeParamsAlias2,
                 )
 
+    def test_evaluate_local_generic(self):
+        class Cls:
+            x: alias
+            y: alias2
+            z: alias3
+
+        fwdref = ForwardRef("alias[int]", owner=Cls)
+        with self.assertRaises(NameError):
+            fwdref.evaluate()
+
+        alias = list
+        self.assertEqual(fwdref.evaluate(), list[int])
```

Comment:
This is incorrect; `alias` is not defined in the class namespace when the class is created.

Reply:
Yes, sorry, should have included nonlocal, etc. The test still passed without it, so didn't consider it before submitting.